src/prognostic_equations/advection.jl

```diff
@@ -422,6 +427,284 @@ function edmfx_sgs_vertical_advection_tendency!(
+            ᶠwₛ³ʲs =
+                (@. lazy(CT3(ᶠinterp(Geometry.WVector(-1 * ᶜwₛʲs.:($$j))))))
+
+            ᶜa = (@. lazy(draft_area(Y.c.sgsʲs.:($$j).ρa, ᶜρʲs.:($$j))))
+
+            # Flux form vertical advection of rho * area with sedimentation contributions
+            vtt = vertical_transport(
+                ᶜρʲs.:($j),
+                ᶠwₗ³ʲs,
+                (@. lazy(ᶜa * Y.c.sgsʲs.:($$j).q_liq)),
+                dt,
+                edmfx_upwinding,
+            )
+            @. Yₜ.c.sgsʲs.:($$j).ρa += vtt
+            vtt = vertical_transport(
+                ᶜρʲs.:($j),
+                ᶠwᵢ³ʲs,
+                (@. lazy(ᶜa * Y.c.sgsʲs.:($$j).q_ice)),
+                dt,
+                edmfx_upwinding,
+            )
+            @. Yₜ.c.sgsʲs.:($$j).ρa += vtt
+            vtt = vertical_transport(
+                ᶜρʲs.:($j),
+                ᶠwᵣ³ʲs,
+                (@. lazy(ᶜa * Y.c.sgsʲs.:($$j).q_rai)),
+                dt,
+                edmfx_upwinding,
+            )
+            @. Yₜ.c.sgsʲs.:($$j).ρa += vtt
+            vtt = vertical_transport(
+                ᶜρʲs.:($j),
+                ᶠwₛ³ʲs,
+                (@. lazy(ᶜa * Y.c.sgsʲs.:($$j).q_sno)),
+                dt,
+                edmfx_upwinding,
+            )
+            @. Yₜ.c.sgsʲs.:($$j).ρa += vtt
+
+            # q_tot and moisture tracers advective form advection with sedimentation velocity
+            va = vertical_advection(
+                ᶠwₗ³ʲs,
+                Y.c.sgsʲs.:($j).q_liq,
+                edmfx_upwinding,
+            )
+            @. Yₜ.c.sgsʲs.:($$j).q_tot += (1 - Y.c.sgsʲs.:($$j).q_tot) * va
+            @. Yₜ.c.sgsʲs.:($$j).q_liq += va
+            va = vertical_advection(
+                ᶠwᵢ³ʲs,
+                Y.c.sgsʲs.:($j).q_ice,
+                edmfx_upwinding,
+            )
+            @. Yₜ.c.sgsʲs.:($$j).q_tot += (1 - Y.c.sgsʲs.:($$j).q_tot) * va
+            @. Yₜ.c.sgsʲs.:($$j).q_ice += va
+            va = vertical_advection(
+                ᶠwᵣ³ʲs,
+                Y.c.sgsʲs.:($j).q_rai,
+                edmfx_upwinding,
+            )
+            @. Yₜ.c.sgsʲs.:($$j).q_tot += (1 - Y.c.sgsʲs.:($$j).q_tot) * va
+            @. Yₜ.c.sgsʲs.:($$j).q_rai += va
+            va = vertical_advection(
+                ᶠwₛ³ʲs,
+                Y.c.sgsʲs.:($j).q_sno,
+                edmfx_upwinding,
+            )
+            @. Yₜ.c.sgsʲs.:($$j).q_tot += (1 - Y.c.sgsʲs.:($$j).q_tot) * va
+            @. Yₜ.c.sgsʲs.:($$j).q_sno += va
+
+            # mse advective form advection with sedimentation velocity
+            (; ᶜΦ) = p.core
+            thp = CAP.thermodynamics_params(params)
+            ᶜmseₗ = (@. lazy(
+                Y.c.sgsʲs.:($$j).q_liq * (
+                    TD.internal_energy_liquid(thp, ᶜtsʲs.:($$j)) +
+                    TD.gas_constant_air(thp, ᶜtsʲs.:($$j)) *
+                    TD.air_temperature(thp, ᶜtsʲs.:($$j)) +
```

Comment:
```suggestion
                    TD.internal_energy_liquid(thp, ᶜtsʲs.:($$j))
```

Reply:
(And I may call this variable qₗmseₗ to be clear but up to you)

Reply:
Also, you have a Iₗ function defined somewhere and you use it in microphysics I think.

Reply:
I just defined `Iₗ` locally in one function. I don't think I can use it here, unless we move it to some abbreviations file.